src/App.tsx

```diff
@@ -1,27 +1,50 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
+import { get5First, getAll, getRedGoods } from './api/goods';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+export const App: React.FC = () => {
+  const initialGoods = [] as Good[];
+  const [goods, setGoods] = useState(initialGoods);
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+  return (
+    <div className="App">
+      <h1>Dynamic list of Goods</h1>
+      <button
+        type="button"
+        data-cy="all-button"
+        onClick={() => {
+          getAll()
+            .then(item => setGoods(item));
+        }}
+      >
+        Load all goods
+      </button>
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+      <button
+        type="button"
+        data-cy="first-five-button"
+        onClick={() => {
+          get5First()
+            .then(item => setGoods(item));
+        }}
+      >
+        Load 5 first goods
+      </button>
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+      <button
+        type="button"
+        data-cy="red-button"
+        onClick={() => {
+          getRedGoods()
+            .then((item) => setGoods(item));
+        }}
+      >
+        Load red goods
+      </button>
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
-
-    <GoodsList goods={[]} />
-  </div>
-);
+      {goods.length && (<GoodsList goods={goods} />)}
```

Comment:
```suggestion
      {!!goods.length && (<GoodsList goods={goods} />)}
```
<img width="415" alt="image" src="https://user-images.githubusercontent.com/66749919/205430396-d1955446-79de-42e6-9b6f-7655196ca9d4.png">